packages/@sanity/components/src/utilities/ScrollContainer.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styles from './styles/ScrollContainer.css'
+
+export default class ScrollContainer extends React.PureComponent {
+  static propTypes = {
+    className: PropTypes.string,
+    children: PropTypes.node,
+    onScroll: PropTypes.func
+  }
+
+  static defaultProps = {
+    onScroll: () => {}
+  }
+
+  getChildContext() {
+    return {
+      getScrollContainer: () => this._scrollContainerElement
+    }
+  }
+
+  static defaultProps = {
+    className: ''
+  }
+
+  componentDidMount() {
+    this._scrollContainerElement.addEventListener('scroll', this.props.onScroll, {passive: true})
```

Comment:
☝️ This listener should be removed on `componentWillUnmount`

Reply:
👍

Reply:
Looking closer at this, I don't think its wise to use `this.props.onScroll` directly as the event handler, as it may change throughout the lifetime of the component, and therefore the removeEventListener call below may pass a different this.props.onScroll (and not remove the listener).

Would you implement a `handleScroll` method that you instead use as a proxy for `this.props.onScroll`?

Reply:
The issue is that if we fallback to document.body it will no pick up the scrollEvent. As far as i know onScroll is supported by  both element and body.